mailbox/store/src/main/java/org/apache/james/mailbox/store/FetchBatchSizes.java

```diff
@@ -0,0 +1,150 @@
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.store;
+
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+
+public class FetchBatchSizes {
+   
+    public static final int DEFAULT_FETCH_BATCH_SIZE = 200;
+
+    public static FetchBatchSizes defaultValues() {
+        return new Builder().build();
+    }
+
+    public static FetchBatchSizes uniqueBatchSize(int batchSize) {
+        return new Builder()
+                .metadata(batchSize)
+                .headers(batchSize)
+                .body(batchSize)
+                .full(batchSize)
+                .build();
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+
+        private Optional<Integer> metadata;
+        private Optional<Integer> headers;
+        private Optional<Integer> body;
+        private Optional<Integer> full;
+
+        private Builder() {
+            metadata = Optional.absent();
+            headers = Optional.absent();
+            body = Optional.absent();
+            full = Optional.absent();
+        }
+
+        public Builder metadata(int fetchBatchSize) {
+            Preconditions.checkState(fetchBatchSize > 0, "'metdata' must be greater then zero");
+            this.metadata = Optional.of(fetchBatchSize);
+            return this;
+        }
+
+        public Builder headers(int fetchBatchSize) {
+            Preconditions.checkState(fetchBatchSize > 0, "'headers' must be greater then zero");
+            this.headers = Optional.of(fetchBatchSize);
+            return this;
+        }
+
+        public Builder body(int fetchBatchSize) {
+            Preconditions.checkState(fetchBatchSize > 0, "'body' must be greater then zero");
+            this.body = Optional.of(fetchBatchSize);
+            return this;
+        }
+
+        public Builder full(int fetchBatchSize) {
+            Preconditions.checkState(fetchBatchSize > 0, "'full' must be greater then zero");
+            this.full = Optional.of(fetchBatchSize);
+            return this;
+        }
+
+        public FetchBatchSizes build() {
+            return new FetchBatchSizes(
+                    metadata.or(DEFAULT_FETCH_BATCH_SIZE),
+                    headers.or(DEFAULT_FETCH_BATCH_SIZE),
+                    body.or(DEFAULT_FETCH_BATCH_SIZE),
+                    full.or(DEFAULT_FETCH_BATCH_SIZE)
```

Comment:
In the past, I have been told that you should not add newline before `);`. Personally I do not mind, but if you want to be coherent with what has been told before remove it